Add audit log common data and PI modification handlers in the Camunda Exporter

koevskinikola opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->

- Add a shared `AuditLogHandler` class in the Camunda Exporter.
- Define a composite transformer interface that modifies the behavior of the `AuditLogHandler` class depending on the record type.
- Implement a Process Instance modification operation transformer for the audit log handler.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

closes #40543

koevskinikola committed:
refactor: adjust webapp schema audit log entity properties

Remove the operation details property as we will evaluate
different approaches in implementing it.

Change the types of several properties to reflect an updated
data model and align with the service layer AuditLogEntity.

koevskinikola committed:
refactor: add unknown enum to audit log enums

Catch unknown operations that are included in the audit log

koevskinikola committed:
feat: add a general audit log handler and a composite interface

Extract all common audit log data from records with
a single AuditLogHandler exporter handler class. Then
modify the behavior of the class using implementations
of the AuditLogOperationTransformer.

koevskinikola committed:
feat: add a pi modification audit log transformer

Add a composite transformer for Process Instance
modifications for the AuditLogHandler and enable
the tracking of PI Modification operations in the
audit log.

koevskinikola committed:
feat: include actual tenant id in pi modification record

Invlude the tenant owner of the process instance that
is modified in the Process Instance Modification event.

Previously, there was no tenantId property, and the
avalable getTenantId() method returned a hardcoded
<default> tenant value.

The background behind this implementation was that
this getter was never used, but we needed to
implement it for the purposes of multi-tenancy.

koevskinikola committed:
test: add test coverage for pi modification audit log handler

igpetrov reviewed (approved):
Besides failing tests, LGTM

igpetrov commented on webapps-schema/src/main/java/io/camunda/webapps/schema/entities/auditlog/AuditLogScope.java:
```diff
@@ -0,0 +1,13 @@
+ */
+package io.camunda.webapps.schema.entities.auditlog;
+
+public enum AuditLogScope {
```
On a second thought, shall we maybe call it `AuditLogTenantScope` to make it more specific?

koevskinikola commented on webapps-schema/src/main/resources/schema/elasticsearch/create/template/camunda-audit-log.json:
```diff
@@ -93,19 +95,6 @@
-      "decisionKey": {
-        "type": "long"
-      },
-      "join": {
```
I had the idea of using it for connecting `BATCH` audit log entries with the `batch item` entries, example:
- `CREATE BATCH` operation for PI migration - this would have been the parent entry
- `PI MIGRATE` operation 1 - this would have been the child entry
- `PI MIGRATE` operation 2 - this would have been the child entry

However, we don't need the JOIN relation, as we don't have a use-case for it. Batch items will be visible only in the Operate PI details view, and we will navigate to the Batch Operation Monitoring UI (different epic) using the `batchOperationKey`.

We won't really query for all of the items. But I will confirm this with the BOM team.

koevskinikola committed:
test: temporarily disable backup restore tests

The BackupRestoreTests currently tests against a
SNAPSHOT docker image with main branch code. This
logic makes it impossible to make changes to
secondary storage schemas, as the Index schema
validator marks them as incompatible changes.

Until the issue is fixed, we need to toggle
these tests to merge any PR changes.
